engine/src/main/java/org/destinationsol/GameOptions.java

```diff
@@ -220,29 +222,14 @@ public GameOptions(boolean mobile, SolFileReader solFileReader) {
     }
 
     public void advanceResolution() {
-        if (resolutionIterator == null) {
-            // Initialize resolution choices - get the resolutions that are supported
-            Graphics.DisplayMode[] displayModes = Gdx.graphics.getDisplayModes();
-
-            for (Graphics.DisplayMode displayMode : displayModes) {
-                supportedResolutions.add(displayMode.width + "x" + displayMode.height);
```

Comment:
This line did that, when several display modes with the same resolution were offered (for instance with different hertzrate), only one was saved, as it was set. Now, for several display modes with the same resolution, each gets registered separatelly, making it appear to the user as nothing happens for a while as he clicks on the advance resolution button.

Reply:
Good catch. Although I believe we can take care of that in increase() method, by making sure `oldX != x || oldY != y`

Reply:
I made the list of Resolutions distinct in the ResolutionProvider.